covalent/_dispatcher_plugins/local.py

```diff
@@ -189,9 +226,63 @@ def func(*new_args, **new_kwargs):
                 dispatch_id, new_args, new_kwargs, replace_electrons, reuse_previous_results
             )
 
-            test_url = f"http://{dispatcher_addr}/api/redispatch"
-            r = requests.post(test_url, json=body)
+            redispatch_url = f"{dispatcher_addr}/api/redispatch"
+            r = requests.post(redispatch_url, json=body, params={"is_pending": is_pending})
             r.raise_for_status()
             return r.content.decode("utf-8").strip().replace('"', "")
 
         return func
+
+    @staticmethod
+    def register_triggers(triggers_data: List[Dict], dispatch_id: str) -> None:
+        """
+        Register the given triggers to the Triggers server.
+        Register also starts the `observe()` method of said trigger.
+        This is done by calling `BaseTrigger._register` method
+        with the given trigger dictionary and is equivalent to
+        calling the trigger objects `register()` method.
+
+        Args:
+            triggers_data: List of trigger dictionaries to be registered
+            dispatch_id: Lattice's dispatch id to be linked with given triggers
+
+        Returns:
+            None
+        """
+
+        for tr_dict in triggers_data:
+            tr_dict["lattice_dispatch_id"] = dispatch_id
+            BaseTrigger._register(tr_dict)
+
+    @staticmethod
+    def stop_triggers(
+        dispatch_ids: Union[str, List[str]], triggers_server_addr: str = None
+    ) -> None:
+        """
+        Stop observing on all triggers of all given dispatch ids registered on the Triggers server.
+        Args:
+            dispatch_ids: Dispatch ID(s) for whose triggers are to be stopped
+            triggers_server_addr: Address of the Triggers server; configured dispatcher's address is used as default
+        Returns:
+            None
+        """
+
+        if triggers_server_addr is None:
+            triggers_server_addr = (
+                "http://"
+                + get_config("dispatcher.address")
+                + ":"
+                + str(get_config("dispatcher.port"))
+            )
+
+        stop_triggers_url = f"{triggers_server_addr}/api/triggers/stop_observe"
+
+        if isinstance(dispatch_ids, str):
+            dispatch_ids = [dispatch_ids]
+
+        r = requests.post(stop_triggers_url, json=dispatch_ids)
+        r.raise_for_status()
+
+        print("Triggers for following dispatch_ids have stopped observing:")
```

Comment:
@kessler-frost can we remove this print statement and adding it in as log statements?

Reply:
Removed 👍